@ironMann : The FullCI fails in datadistribution probably due to the o2::byte / std::byte change:
```
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/DataDistribution/v0.10.16/v0.10.16/src/common/SubTimeFrameFileReader.cxx: In member function 'o2::header::Stack o2::DataDistribution::SubTimeFrameFileReader::getHeaderStack(std::size_t&)':
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/DataDistribution/v0.10.16/v0.10.16/src/common/SubTimeFrameFileReader.cxx:121:30: error: cannot convert 'unsigned char*' to 'std::byte*' in initialization
```
Could you have a look? (This ran yesterday evening)

--- a/CCDB/src/CcdbApi.cxx
+++ b/CCDB/src/CcdbApi.cxx
@@ -73,6 +73,7 @@ void CcdbApi::init(std::string const& host)
 
   // find out if we can can in principle connect to Alien
   mHaveAlienToken = checkAlienToken();
+  LOG(INFO) << "WITH ALIEN TOKEN?: " << mHaveAlienToken;
 }
 
 /**
@@ -468,6 +469,12 @@ void* CcdbApi::extractFromLocalFile(std::string const& filename, std::type_info 
 
 bool CcdbApi::checkAlienToken() const
 {
+#ifdef __APPLE__
+  // not checking for token on Mac because
+  // a) we have seen problems where system call below hangs in some cases
+  // b) not the production plattform where the token would be beneficial
+  return false;
+#endif
   // a somewhat weird construction to programmatically find out if we
   // have a GRID token; Can be replaced with something more elegant once
   // alien-token-info does not ask for passwords interactively
@@ -475,6 +482,9 @@ bool CcdbApi::checkAlienToken() const
     return true;
   }
   auto returncode = system("alien-token-info > /dev/null 2> /dev/null");
+  if (returncode == -1) {
+    LOG(ERROR) << "system(\"alien-token-info\") call failed with internal fork/wait error";
+  }
   return returncode == 0;
 }
 

--- a/run/O2SimDeviceRunner.cxx
+++ b/run/O2SimDeviceRunner.cxx
@@ -35,20 +35,19 @@ namespace bpo = boost::program_options;
 std::vector<int> gChildProcesses; // global vector of child pids
 int gMasterProcess = -1;
 
-// custom signal handler to ensure that we
-// distribute signals to all detached forks
-void sighandler(int signal)
+void sigaction_handler(int signal, siginfo_t* signal_info, void*)
 {
-  if (signal == SIGTERM || signal == SIGINT) {
-    auto pid = getpid();
-    if (pid == gMasterProcess) {
-      // the master
-      for (auto child : gChildProcesses) {
-        kill(child, signal);
-      }
-    }
-    _exit(0);
+  auto pid = getpid();
+  LOG(INFO) << pid << " caught signal from source " << signal_info->si_pid;
+  auto groupid = getpgrp();
+  if (pid == gMasterProcess) {
+    killpg(pid, signal); // master kills whole process group
+  } else {
+    // forward to master
+    kill(groupid, signal);
   }
+  _exit(1); // we treat signal interruption as an error
+            // because only ordinary termination is good in the context of the distributed system
 }
 
 void addCustomOptions(bpo::options_description& options)
@@ -226,14 +225,20 @@ bool waitForControlInput()
 
 int main(int argc, char* argv[])
 {
-  // enable signal handler for termination signals
-  if (signal(SIGTERM, sighandler) == SIG_IGN) {
-    signal(SIGTERM, SIG_IGN);
+  struct sigaction act;
+  memset(&act, 0, sizeof act);
+  sigemptyset(&act.sa_mask);
+  act.sa_sigaction = &sigaction_handler;
+  act.sa_flags = SA_SIGINFO; // <--- enable sigaction
+
+  std::vector<int> handledsignals = {SIGTERM, SIGINT, SIGQUIT, SIGSEGV, SIGBUS, SIGFPE}; // <--- may need to be completed
+  // remember that SIGKILL can't be handled
+  for (auto s : handledsignals) {
+    if (sigaction(s, &act, nullptr)) {
+      LOG(ERROR) << "Could not install signal handler for " << s;
+      exit(EXIT_FAILURE);
+    }
   }
-  if (signal(SIGINT, sighandler) == SIG_IGN) {
-    signal(SIGINT, SIG_IGN);
-  }
-  signal(SIGCHLD, SIG_IGN); /* Silently reap children to avoid zombies. */
 
   // extract the path to FairMQ config
   bpo::options_description desc{"Options"};

--- a/run/o2sim_parallel.cxx
+++ b/run/o2sim_parallel.cxx
@@ -69,7 +69,7 @@ void remove_tmp_files()
   // remove all (known) socket files in /tmp
   // using the naming convention /tmp/o2sim-.*PID
   std::stringstream command;
-  command << "rm /tmp/o2sim-*" << getpid();
+  command << "rm /tmp/o2sim-* " << getpid() << " 2>/dev/null";
   auto rc = system(command.str().c_str()); // a solution based on std::filesystem may be preferred but is longer
 }
 
@@ -167,7 +167,7 @@ void sighandler(int signal)
     cleanup();
     // forward signal to all children
     for (auto& pid : gChildProcesses) {
-      kill(pid, signal);
+      killpg(pid, signal);
     }
     exit(0);
   }
@@ -446,6 +446,7 @@ int main(int argc, char* argv[])
     return r;
   } else {
     gChildProcesses.push_back(pid);
+    setpgid(pid, pid);
     close(pipe_serverdriver_fd[1]);
     std::cout << "Spawning particle server on PID " << pid << "; Redirect output to " << getServerLogName() << "\n";
 
@@ -487,6 +488,7 @@ int main(int argc, char* argv[])
       return 0;
     } else {
       gChildProcesses.push_back(pid);
+      setpgid(pid, pid); // the worker processes will form their own group
       std::cout << "Spawning sim worker " << id << " on PID " << pid
                 << "; Redirect output to " << workerlogss.str() << "\n";
     }
@@ -514,6 +516,7 @@ int main(int argc, char* argv[])
     return 0;
   } else {
     std::cout << "Spawning hit merger on PID " << pid << "; Redirect output to " << getMergerLogName() << "\n";
+    setpgid(pid, pid);
     gChildProcesses.push_back(pid);
     close(pipe_mergerdriver_fd[1]);
 
@@ -529,7 +532,7 @@ int main(int argc, char* argv[])
         if (!conf.asService()) {
           LOG(INFO) << "SIMULATION IS DONE. INITIATING SHUTDOWN.";
           for (auto p : gChildProcesses) {
-            kill(p, SIGTERM);
+            killpg(p, SIGTERM);
           }
         } else {
           LOG(INFO) << "SIMULATION DONE. STAYING AS DAEMON.";
@@ -546,19 +549,19 @@ int main(int argc, char* argv[])
   int status, cpid;
   // wait just blocks and waits until any child returns; but we make sure to wait until merger is here
   while ((cpid = wait(&status)) != mergerpid) {
-    if (WIFSIGNALED(status)) {
+    if (WEXITSTATUS(status) || WIFSIGNALED(status)) {
       LOG(INFO) << "Process " << cpid << " EXITED WITH CODE " << WEXITSTATUS(status) << " SIGNALED "
                 << WIFSIGNALED(status) << " SIGNAL " << WTERMSIG(status);
-    }
-    // we bring down all processes if one of them aborts
-    if (WTERMSIG(status) == SIGABRT) {
-      LOG(INFO) << "Problem detected ";
+
+      // we bring down all processes if one of them had problems or got a termination signal
+      // if (WTERMSIG(status) == SIGABRT || WTERMSIG(status) == SIGSEGV || WTERMSIG(status) == SIGBUS || WTERMSIG(status) == SIGTERM) {
+      LOG(INFO) << "Problem detected (or child received termination signal) ... shutting down whole system ";
       for (auto p : gChildProcesses) {
         LOG(INFO) << "KILLING " << p;
-        kill(p, SIGTERM);
+        killpg(p, SIGTERM); // <--- makes sure to shutdown "unknown" child pids via the group property
       }
       cleanup();
-      LOG(FATAL) << "ABORTING DUE TO ABORT IN COMPONENT";
+      LOG(ERROR) << "SHUTTING DOWN DUE TO SIGNALED EXIT IN COMPONENT " << cpid;
     }
   }
   // This marks the actual end of the computation (since results are available)
@@ -569,7 +572,7 @@ int main(int argc, char* argv[])
   for (auto p : gChildProcesses) {
     if (p != mergerpid) {
       LOG(INFO) << "SHUTTING DOWN CHILD PROCESS " << p;
-      kill(p, SIGTERM);
+      killpg(p, SIGTERM);
     }
   }
 

--- a/run/o2sim_parallel.cxx
+++ b/run/o2sim_parallel.cxx
@@ -69,7 +69,7 @@ void remove_tmp_files()
   // remove all (known) socket files in /tmp
   // using the naming convention /tmp/o2sim-.*PID
   std::stringstream command;
-  command << "rm /tmp/o2sim-* " << getpid() << " 2>/dev/null";
+  command << "rm /tmp/o2sim-*-" << getpid() << " 2>/dev/null";
   auto rc = system(command.str().c_str()); // a solution based on std::filesystem may be preferred but is longer
 }
 

--- a/run/o2simtopology_template.json
+++ b/run/o2simtopology_template.json
@@ -10,7 +10,7 @@
                             {
                                 "type":"req",
                                 "method":"connect",
-                                "address":"ipc:///tmp/o2sim-primary-get_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primary-get-#PID#",
                                 "sndBufSize":"100000",
                                 "rcvBufSize":"100000",
                                 "rateLogging":"0"
@@ -23,7 +23,7 @@
                             {
                                 "type":"req",
                                 "method":"connect",
-                                "address":"ipc:///tmp/o2sim-primserv-info_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primserv-info-#PID#",
                                 "sndBufSize":"1000",
                                 "rcvBufSize":"1000",
                                 "rateLogging":"0"
@@ -36,7 +36,7 @@
                             {
                                 "type":"push",
                                 "method":"connect",
-                                "address":"ipc:///tmp/o2sim-hitmerger-simdata_25009#PID#",
+                                "address":"ipc:///tmp/o2sim-hitmerger-simdata-#PID#",
                                 "sndBufSize":1000,
                                 "rcvBufSize":1000,
                                 "rateLogging":0
@@ -54,7 +54,7 @@
                             {
                                 "type":"rep",
                                 "method":"bind",
-                                "address":"ipc:///tmp/o2sim-primary-get_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primary-get-#PID#",
                                 "sndBufSize":100000,
                                 "rcvBufSize":100000,
                                 "rateLogging":0
@@ -67,7 +67,7 @@
                             {
                                 "type":"rep",
                                 "method":"bind",
-                                "address":"ipc:///tmp/o2sim-primserv-info_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primserv-info-#PID#",
                                 "sndBufSize":1000,
                                 "rcvBufSize":1000,
                                 "rateLogging":0
@@ -98,7 +98,7 @@
                             {
                                 "type":"req",
                                 "method":"connect",
-                                "address":"ipc:///tmp/o2sim-primary-get_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primary-get-#PID#",
                                 "sndBufSize":"1000",
                                 "rcvBufSize":"1000",
                                 "rateLogging":"0"
@@ -111,7 +111,7 @@
                             {
                                 "type":"req",
                                 "method":"connect",
-                                "address":"ipc:///tmp/o2sim-primserv-info_25005#PID#",
+                                "address":"ipc:///tmp/o2sim-primserv-info-#PID#",
                                 "sndBufSize":"1000",
                                 "rcvBufSize":"1000",
                                 "rateLogging":"0"
@@ -124,7 +124,7 @@
                             {
                                 "type":"pull",
                                 "method":"bind",
-                                "address":"ipc:///tmp/o2sim-hitmerger-simdata_25009#PID#",
+                                "address":"ipc:///tmp/o2sim-hitmerger-simdata-#PID#",
                                 "sndBufSize":1000,
                                 "rcvBufSize":1000,
                                 "rateLogging":0

